src/sql/workbench/contrib/queryplan2/browser/queryPlan.ts

```diff
@@ -116,6 +116,14 @@ export class QueryPlan2View implements IPanelView {
 				this._qps.push(qp2);
 				this._graphs.push(g);
 				this.updateRelativeCosts();
+
+				// If there is only one query plan, set the height of the query plan to 100%
+				// Later on, if other plans are added we decrease the height to 500px
+				if (this._qps.length === 1) {
+					this._qps[0].container.style.height = '100%';
+				} else if (this._qps.length > 1) {
+					this._qps[0].container.style.height = '500px';
```

Comment:
Where is the 500px value coming from? And why are you only setting the height of the first to 500px when there's more than one?

I'm not sure that this will look nice in all situations too - what happens if the user expands the panel to be >1000px high and has two plans? Will you end up with a gap again? What might be better is to look into making each plan take up an equal % of the total height, with a minimum height of 500px or whatever seems reasonable so that they can expand to fill the space as needed.